tests/phpunit/api/v3/MembershipTest.php

```diff
@@ -189,97 +189,86 @@ public function testActivityForCancelledContribution(): void {
-      'membership_id' => $membershipID2,
+      'line_items' => [
+        [
+          'line_item' => [
+            [
+              'membership_type_id' => 'General',
+              'source' => 'Webform Payment',
+              'line_total' => 50,
+              'qty' => 1,
+            ],
+          ],
+          'params' => [
+            'contact_id' => $contactId1,
+            'membership_type_id' => 'General',
+          ],
+        ],
+        [
+          'line_item' => [
+            [
+              'membership_type_id' => 'General',
+              'source' => 'Webform Payment',
+              'line_total' => 50,
+              'qty' => 1,
+            ],
+          ],
+          'params' => [
+            'contact_id' => $contactId2,
+            'membership_type_id' => 'General',
+          ],
+        ],
+        [
+          'line_item' => [
+            [
+              'membership_type_id' => 'General',
+              'source' => 'Webform Payment',
+              'line_total' => 50,
+              'qty' => 1,
+            ],
+          ],
+          'params' => [
+            'contact_id' => $contactId3,
+            'membership_type_id' => 'General',
+          ],
+        ],
+      ],
     ]);
 
-    // Contact 3
-    $contactId3 = $this->individualCreate();
-    $membershipParam['contact_id'] = $contactId3;
-    $membershipID3 = $this->contactMembershipCreate($membershipParam);
-    $this->callAPISuccess('MembershipPayment', 'create', [
-      'sequential' => 1,
-      'contribution_id' => $contribution['id'],
-      'membership_id' => $membershipID3,
-    ]);
     $_REQUEST['action'] = 'update';
-    $_REQUEST['id'] = $contribution['id'];
+    $_REQUEST['id'] = $order['id'];
     // Change Payment Status to Completed
     /* @var CRM_Contribute_Form_Contribution $form */
     $form = $this->getFormObject('CRM_Contribute_Form_Contribution', [
-      'total_amount' => 100,
+      'total_amount' => 150,
       'financial_type_id' => '1',
       'contact_id' => $contactId1,
       'payment_instrument_id' => CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'payment_instrument_id', 'Check'),
       'contribution_status_id' => 1,
     ]);
     $form->buildForm();
     $form->postProcess();
+    $memberships = $this->callAPISuccess('Membership', 'get', ['sort' => 'contact_id', 'sequential' => 1])['values'];
 
-    // check for Membership 1
-    $params = ['id' => $membershipID1, 'return' => ['contact_id', 'status_id']];
-    $membership1 = $this->callAPISuccess('Membership', 'get', $params);
-    $result1 = $membership1['values'][$membershipID1];
-    $this->assertEquals($result1['contact_id'], $contactId1);
-    $this->assertEquals($result1['status_id'], $newMembershipId);
+    $this->assertEquals($contactId1, $memberships[0]['contact_id']);
+    $this->assertEquals('test status', CRM_Core_PseudoConstant::getName('CRM_Member_BAO_Membership', 'status_id', $memberships[0]['status_id'], 'New'));
```

Comment:
I think this PR is ok but the `'New'` was confusing me, but it seems getName() only takes 3 params so the New does nothing. Was it meant to check that the membership is in a New state?

Reply:
Good catch. Can you fix this @eileenmcnaughton and then it can be MOP.

Reply:
@colemanw @demeritcowboy fixed - thanks!